panels/user-accounts/cc-app-permissions.c

```diff
@@ -433,7 +434,18 @@ update_allow_app_installation (CcAppPermissions *self)
 static void
 setup_parental_control_settings (CcAppPermissions *self)
 {
+  g_autoptr(GDBusConnection) system_bus = NULL;
   gboolean is_authorized, user_is_administrator;
+  g_autoptr(GError) error = NULL;
+
+  system_bus = g_bus_get_sync (G_BUS_TYPE_SYSTEM, self->cancellable, &error);
+  if (system_bus == NULL)
+    {
+      g_warning ("Error getting system bus for setting up parental controls: %s", error->message);
+      return;
+    }
+
+  self->manager = mct_manager_new (system_bus);
```

Comment:
I suspect this line of code might get executed multiple times (`setup_parental_control_settings()` gets called from a few places) and hence will leak old `manager` instances. You should only need to construct the `manager` once, when the panel is first instantiated.

Reply:
> You should only need to construct the manager once, when the panel is first instantiated.

Do you mean when "Users" panels is instantiated (in `panels/user-accounts/cc-user-panel.c`) or when the widget CcAppPermissions is instantiated (in `panels/user-accountscc-app-permissions.c`) ?

Reply:
Probably the latter, but whatever makes the most sense to you. I think they probably are both instantiated at roughly the same time?

Reply:
Right, addressed in the latest iteration of the patch. Picked up the latter approach as that was most sensible to me. Thanks for this pointer here.